### DISCUSSION TITLE
WSL - wine cannot access to the host path DLL files

### DISCUSSION THREAD

**Discussion by D3vil0p3r** [microsoft/WSL#9861] [Q&A]
### Windows Version

Microsoft Windows [Version 10.0.22621.1265]

### WSL Version

1.1.3.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.15.90.1-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) 

### Distro Version

Arch Linux latest

### Other Software

wine-8.4

### Repro Steps

Hello,
I'm using a WSL image of Arch Linux in Windows 11 and, since the WSL can access to host folders, the PATH env variable on Arch WSL image has also System32 path. Usually, in WSL images, the C drive is accessed by `/mnt/c` directory.

I'm trying to use wine for running Remote Desktop Service of the Windows host (so, the `mstsc.exe` executable).

I noted that, when I try to run the Remote Desktop Service by "mstsc.exe" or "wine mstsc.exe" I get the following output:
```
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0108:err:module:import_dll Library api-ms-win-core-commandlinetoargv-l1-1-0.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\shcore.dll") not found
0108:err:module:import_dll Library shcore.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\SHLWAPI.dll") not found
0108:err:module:import_dll Library SHLWAPI.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\mstsc.exe") not found
0108:err:module:import_dll Library api-ms-win-core-commandlinetoargv-l1-1-0.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\SHCORE.dll") not found
0108:err:module:import_dll Library SHCORE.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\mstsc.exe") not found
0108:err:module:LdrInitializeThunk Importing dlls for L"Z:\\mnt\\c\\Windows\\system32\\mstsc.exe" failed, status c0000135
```
despite these dll files exist:
```
ls /mnt/c/Windows/System32/shlwapi.dll
 /mnt/c/Windows/System32/shlwapi.dll

ls /mnt/c/Windows/System32/SHCore.dll
 /mnt/c/Windows/System32/SHCore.dll
```
I tried also to delete `$HOME/.wine` and run `env WINEPREFIX=$HOME/.wine wine /mnt/c/Windows/System32/mstsc.exe` but I get the same error.

Of course, on Windows host, I can run "mstsc.exe" by CMD with no issues.

It seems that on a WSL container, Wine cannot have access to the dependencies/DLLs on the Windows host by the Windows paths mapped on WSL.

For reproducing this:
1. Install an Arch Linux WSL image
2. Install `wine`
3. Run `wine mstsc.exe`

### Expected Behavior

`wine mstsc.exe` should show Remote Desktop Service Windows application.

### Actual Behavior

Receive the following output:
```
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0108:err:module:import_dll Library api-ms-win-core-commandlinetoargv-l1-1-0.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\shcore.dll") not found
0108:err:module:import_dll Library shcore.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\SHLWAPI.dll") not found
0108:err:module:import_dll Library SHLWAPI.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\mstsc.exe") not found
0108:err:module:import_dll Library api-ms-win-core-commandlinetoargv-l1-1-0.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\SHCORE.dll") not found
0108:err:module:import_dll Library SHCORE.dll (which is needed by L"Z:\\mnt\\c\\Windows\\system32\\mstsc.exe") not found
0108:err:module:LdrInitializeThunk Importing dlls for L"Z:\\mnt\\c\\Windows\\system32\\mstsc.exe" failed, status c0000135
```

**Comment by ghost** [comment]
I'm entirely befuddled. Why would you want to use wine instead of interop?

**Comment by D3vil0p3r** [comment]
> I'm entirely befuddled. Why would you want to use wine instead of interop?

How can I use interop for calling `mstsc.exe`?

**Comment by ghost** [comment]
If the software you've installed hasn't changed binfmt registrations then just `mstsc.exe` should work. Else it can be done explicitly like `/init /mnt/c/WINDOWS/system32/mstsc.exe mstsc.exe`

**Comment by D3vil0p3r** [comment]
> If the software you've installed hasn't changed binfmt registrations then just `mstsc.exe` should work. Else it can be done explicitly like `/init /mnt/c/WINDOWS/system32/mstsc.exe mstsc.exe`

I didn't do any changes on binfmt registrations. If I run just `mstsc.exe` I get:
```
Cannot open assembly '/mnt/c/Windows/system32/mstsc.exe': File does not contain a valid CIL image.
```

**Comment by ghost** [comment]
Like I said: "If the software you've installed..."
The software you've installed has. Please see: https://github.com/microsoft/WSL/discussions/9755

**Reply by D3vil0p3r** [reply]
https://github.com/microsoft/WSL/discussions/9755 suggests to use `sudo update-binfmts --disable cli` but Arch Linux has not `update-binfmts` command.
